### PR TITLE
DBZ-3340: Adding timestamp in transaction metadata

### PR DESCRIPTION
**Pull request by nitin456**

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/pipeline/txmetadata/TransactionMonitor.java [hunk 1]
```diff
@@ -160,6 +179,7 @@ private void beginTransaction(OffsetContext offsetContext) throws InterruptedExc
         final Struct value = new Struct(TRANSACTION_VALUE_SCHEMA);
         value.put(DEBEZIUM_TRANSACTION_STATUS_KEY, TransactionStatus.BEGIN.name());
         value.put(DEBEZIUM_TRANSACTION_ID_KEY, offsetContext.getTransactionContext().getTransactionId());
+        value.put(DEBEZIUM_TRANSACTION_TIMESTAMP_KEY, offsetContext.getTransactionContext().getTransactionStartTime());
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/pipeline/txmetadata/TransactionMonitor.java:182]: Should this be inserted only when the timestamp is really provided?

File: debezium-core/src/main/java/io/debezium/pipeline/txmetadata/TransactionMonitor.java [hunk 2]
```diff
@@ -133,6 +135,15 @@ public void transactionComittedEvent(OffsetContext offset) throws InterruptedExc
         offset.getTransactionContext().endTransaction();
     }
 
+    public void transactionComittedEvent(OffsetContext offset, long transactionEndTime) throws InterruptedException {
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/pipeline/txmetadata/TransactionMonitor.java:138]: This and the other new method are almost same as methods without `transactionEndTime`. Could you please refactor the code so the methods ae not copied but one calls another?

File: debezium-core/src/main/java/io/debezium/pipeline/txmetadata/TransactionContext.java [hunk 3]
```diff
@@ -38,9 +38,13 @@
     private final Map<String, Long> perTableEventCount = new HashMap<>();
     private final Map<String, Long> viewPerTableEventCount = Collections.unmodifiableMap(perTableEventCount);
     private long totalEventCount = 0;
+    private long transactionStartTime = 0;
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/pipeline/txmetadata/TransactionContext.java:41]: I'd recommend to use an invalid value like `-1` to indicate unset value as `0` is theaoretically acceptable value.

**Review by jpechane — COMMENTED** [review]
Thanks for the PR. I've added few comments. Please fix the commit message.
Also tests needs to be provided. Other than the it is good to go.

File: debezium-core/src/main/java/io/debezium/pipeline/txmetadata/TransactionContext.java (hunk 3 above)
Comment (nitin456) [reply debezium-core/src/main/java/io/debezium/pipeline/txmetadata/TransactionContext.java:41]: @jpechane Thanks for the review, I will address review comment this week.

**Comment by jpechane** [comment]
Closing as this was implemented via https://issues.redhat.com/browse/DBZ-5170

**Pull request closed without merging**